src/association/association_internal.rs

```diff
@@ -712,13 +714,28 @@ impl AssociationInternal {
             ..Default::default()
         };
 
+        let report_unrecognized_params = i.params.iter().filter_map(|param| {
```

Comment:
I'm a bit confused as to why we say it's "unrecognized" if we know for sure it's a "report"

Reply:
Hi, I think my naming here is sub-optimal. The "report" isn't about the type, but meant as the verb "report this unknown parameter". I'll think about better names.